library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -166,14 +166,20 @@ impl Command {
             fn clone3(cl_args: *mut clone_args, len: libc::size_t) -> libc::c_long
         }
 
+        // Bypassing libc for `clone3` can make further libc calls unsafe,
+        // so we use it sparingly for now. See #89522 for details.
+        // Some tools (e.g. sandboxing tools) may also expect `fork`
+        // rather than `clone3`.
+        let want_clone3 = self.get_create_pidfd();
+
         // If we fail to create a pidfd for any reason, this will
         // stay as -1, which indicates an error.
         let mut pidfd: pid_t = -1;
 
         // Attempt to use the `clone3` syscall, which supports more arguments
         // (in particular, the ability to create a pidfd). If this fails,
         // we will fall through this block to a call to `fork()`
-        if HAS_CLONE3.load(Ordering::Relaxed) {
+        if want_clone3 && HAS_CLONE3.load(Ordering::Relaxed) {
             let mut flags = 0;
             if self.get_create_pidfd() {
```

Comment:
This conditional is now redundant.

Reply:
It's only redundant if you assume that `want_clone3` will never encompass anything else.

Reply:
OK, I've renamed the flag to narrow its scope, then removed the inner conditional.